README.md

```diff
@@ -434,9 +434,17 @@ You can find an example in [idam-web-public](https://github.com/hmcts/idam-web-p
 
 The current state of the Nightly Pipeline is geared towards testing both frontend and backend applications served by NodeJS, AngularJS and Java APIs.
 
-The pipeline will automatically detect whether you're application is node based or gradle based and run the appropriate security tests based on that.
+The pipeline will automatically detect whether your application is node based or gradle based and run the appropriate security tests based on that.
 
-If you have a requirement to customise the script, you can place your own script in a folder called `ci` in your repo. Make sure to call the script `security.sh`.
+Gradle based applications are more commonly used in the backend but if your frontend application is gradle based, you can pass a parameter to `withNightlyPipeline` to indicate this is the case and run the frontend specific security script instead of the default backend specific script. This value is optional.
+
+```
+def isFrontend = true
+
+withNightlyPipeline(type, product, component, isFrontend)
```

Comment:
no we shouldn't modify the parameters, they are fixed for all pipelines at:
`type, product, component`

in the DSL callbacks a flag can be added, something like:

```
withNightlyPipeline(type, product, component) {
  useFrontendSecurityTests()
}
```

Reply:
Would that be a case of adding a new function to the dsl like this?
```
  void useFrontendSecurityTests(String urlExclusions = "", int timeout = 120) {
    config.urlExclusions = urlExclusions
    config.securityScanTimeout = timeout
    config.useFrontendSecurityTests = true
  }
```

And then new logic to `sectionNightlyTests` where the variable would be set as required?
```
    if (config.useFrontendSecurityTests) {
      stageWithAgent('Security scan', product) {
        env.IS_FRONTEND = "true"
        warnError('Failure in securityScan') {
          pcr.callAround('securityScan') {
            timeout(time: config.securityScanTimeout, unit: 'MINUTES') {
              builder.securityScan()
            }
          }
        }
      }
    }
```

Reply:
um no not really, we can pair tomorrow to finish this off